### PR TITLE
RD-2810 Show deployment name in translations

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: https://cloudifysource.atlassian.net/browse/RD-2810

This ticket is about showing the deployment name in addition to deployment id in the following places:
1. Execute Workflow Modal L232
2. Execute Workflow Modal L104
3. Labels Modal L87 that is called from:
    A.  AddLabelsModal
    B.  ManageLabelsModal

The update in Blueprint Deployment is done in a separate PR: https://github.com/cloudify-cosmo/cloudify-stage/pull/1916

I also fixed a typo "Deplyment" to "Deployment".

## Screenshots / Videos
ExecuteWorkflowModal.tsx L232:
Screen Shot 2022-04-18 at 10 21 56

LabelsModal.tsx L87:
Screen Shot 2022-04-19 at 13 24 16

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

19 April 2022 running `blueprints_spec`:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2559/pipeline

19 April 2022 running all system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2555/pipeline
Result: all tests except `blueprints_spec` passed.

17 April 2022:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2546/pipeline

Initial:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2471/pipeline

 

## Documentation
https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2175

### CODE REVIEW

**Comment by leon-good-life** [comment]
In DeploymentActionsMenu, item.name is taken from menuItems variable which is a hardcoded object without an id. I think maybe it should not show the deployment id there. And the ticket is about showing deployment name which is there already.

I think I need to find more places in the code in which I may need to show deployment name.

**Comment by leon-good-life** [comment]
Things left to do:
1. Check if there are other places in which I need to show deployment name.
2. Check and update the documentation.
3. Verify tests are passing.
4. Add screenshots to the PR description.

File: app/translations/en.json [hunk 1]
```diff
@@ -528,14 +528,14 @@
                 "execute": {
                     "header": {
                         "noDeployment": "Execute workflow {{workflowName}}",
-                        "singleDeployemnt": "Execute workflow {{workflowName}} on {{deploymentId}}",
-                        "singleNamedDeployemnt": "Execute workflow {{workflowName}} on {{ deploymentName }}",
+                        "singleDeployment": "Execute workflow {{workflowName}} on {{deploymentName}}",
+                        "singleNamedDeployment": "Execute workflow {{workflowName}} on {{ deploymentName }}",
```
Comment (kubama) [inline app/translations/en.json:532]: These two labels are same, maybe one of them could be removed?

File: widgets/common/src/executeWorkflow/ExecuteWorkflowModal.tsx [hunk 2]
```diff
@@ -100,12 +100,11 @@ const ExecuteWorkflowModal: FunctionComponent<ExecuteWorkflowModalProps> = ({
...
                     } else {
-                        setErrors(
-                            t('errors.workflowError', {
-                                deploymentId,
-                                workflowName
-                            })
-                        );
+                        const deploymentNameAndId = Stage.Utils.formatDisplayName({
+                            id: _.head(deploymentsList),
+                            displayName: deploymentName
```
Comment (qooban) [inline widgets/common/src/executeWorkflow/ExecuteWorkflowModal.tsx:105]: What is the value of `deploymentName` when we have `deploymentsList` with multiple deployments?

File: app/translations/en.json (hunk 1 above)
Comment (leon-good-life) [reply app/translations/en.json]: `singleDeployment` was showing the deployment name correctly before my recent changes. I reverted `singleDeployment` as it was before my changes: [e1d16ef](https://github.com/cloudify-cosmo/cloudify-stage/pull/1897/commits/e1d16ef6e0f682c539c524afcd6e1b064935deed)

File: widgets/common/src/executeWorkflow/ExecuteWorkflowModal.tsx (hunk 2 above)
Comment (leon-good-life) [reply widgets/common/src/executeWorkflow/ExecuteWorkflowModal.tsx:105]: [dd6574d](https://github.com/cloudify-cosmo/cloudify-stage/pull/1897/commits/dd6574d104f69c342ff0ca159c55104c59d26f78)

**Comment by leon-good-life** [comment]
I started system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2546/pipeline

**Comment by leon-good-life** [comment]
All system tests passed

**Comment by leon-good-life** [comment]
I started system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/2555/pipeline

File: app/translations/en.json [hunk 3]
```diff
@@ -669,7 +668,7 @@
                 },
                 "fetchingLabelsError": "Cannot fetch labels for deployment '{{deploymentId}}'",
                 "labelDuplicationError": "Cannot add the same label twice",
-                "modalHeader": "Manage labels for deployment {{ deploymentName }}",
+                "modalHeader": "Manage labels for deployment {{ deploymentNameAndId }}",
```
Comment (leon-good-life) [inline app/translations/en.json:671]: Explanation: It is used in `ManageLabelsModal` which is used in `LabelsModal`.

File: app/translations/en.json [hunk 4]
```diff
@@ -1419,7 +1418,7 @@
             "noLabels": "There are no Labels defined",
             "add": "Add",
             "export": "Export",
-            "addHeader": "Add labels for deployment '{{deploymentId}}'",
+            "addHeader": "Add labels for deployment '{{deploymentNameAndId}}'",
```
Comment (leon-good-life) [inline app/translations/en.json:1421]: **Explanation:** it is used in `AddLabelsModal` which calls `LabelsModal`

**Review by kubama — APPROVED** [review]

**Pull request merged**